pkg/kubelet/kubelet.go

```diff
@@ -1916,25 +1906,28 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 	updated := false
 	for i := range node.Status.Conditions {
 		if node.Status.Conditions[i].Type == api.NodeReady {
-			newCondition.LastTransitionTime = node.Status.Conditions[i].LastTransitionTime
-			if node.Status.Conditions[i].Status != api.ConditionTrue {
-				kl.recordNodeOnlineEvent()
-			}
-			node.Status.Conditions[i] = newCondition
+			oldReadyNodeConditionStatus = node.Status.Conditions[i].Status
```

Comment:
I think that:

    newReadyNodeCondition.LastUpdateTime = currentTime

is not needed as it's already set above (`LastHeartbeatTime`)

Also about: `// also generate appropriate event`, I prefer to keep the logic in just one place (below).